src/rulesets/Base/fastmath_able.jl

```diff
@@ -164,14 +164,16 @@ let
         @scalar_rule x - y (true, -1)
         @scalar_rule x / y (one(x) / y, -(Ω / y))
         #log(complex(x)) is required so it gives correct complex answer for x<0
-        @scalar_rule(x ^ y,
-            (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(complex(x))),
-        )
+        @scalar_rule(x ^ y, (
+            ifelse(iszero(x), ifelse(isone(y), one(Ω), zero(Ω)), y * Ω / x),
+            Ω * log(complex(x)),
+        ))
         # x^y for x < 0 errors when y is not an integer, but then derivative wrt y
         # is undefined, so we adopt subgradient convention and set derivative to 0.
-        @scalar_rule(x::Real ^ y::Real,
-            (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(oftype(Ω, ifelse(x ≤ 0, one(x), x)))),
-        )
+        @scalar_rule(x::Real ^ y::Real, (
+            ifelse(iszero(x), ifelse(isone(y), one(Ω), zero(Ω)), y * Ω / x),
```

Comment:
Are we sure this is the best way to compute this? Division is quite expensive. Why not the more straightforward:
```
function rrule(::typeof(^), x::Real, y::Real)
    r = x^(y-1)
    r*x, Δ->(..., r, ...)
end
```

Reply:
> Are we sure this is the best way to compute this?

If I had to guess, it's so we can use `@scalar_rule` and avoid defining both the `frule` and `rrule`, but I agree that this approach is probably better. For the cases I can think of where this doesn't work (e.g. `x<0` and `y=0.5`), the primal function would itself error.

Reply:
FWIW, I started trying to do that (never dividing) on this branch:

https://github.com/JuliaDiff/ChainRules.jl/compare/master...mcabbott:fix_pow_mine?expand=1

Built on top of this PR, could perhaps push here, or make a separate PR... but still needs many tests & some details perhaps to think about.

Reply:
Sorry, I lost track of this PR a little. If you want to take this over, feel free to either push to this branch or open a new PR. Otherwise, I will try to get back to this some time this week.

Reply:
Sure, either way. I have a lot of `NaN`s still... some of them marked broken tests now, at least.